src/Ogre_glTF_textureImporter.cpp

```diff
@@ -74,13 +76,27 @@ void textureImporter::loadTexture(const tinygltf::Texture& texture)
 											   1,
 											   1,
 											   pixelFormat,
-											   Ogre::TU_DEFAULT);
+											   Ogre::TU_DEFAULT,
+											   nullptr,
+											   IsHwGammaEnabled());
 
 	OgreTexture->loadImage(OgreImage);
 
 	loadedTextures.insert({ texture.source, OgreTexture });
 }
 
+bool textureImporter::IsHwGammaEnabled() const
+{
+	bool isHwGammaEnabled = false;
+
+	auto renderTargetIt = Ogre::Root::getSingleton().getRenderSystem()->getRenderTargetIterator();
+	if(renderTargetIt.hasMoreElements())
```

Comment:
Isn't there any simple solution to check if hardware gamma correction was enabled?

And if we are going the "iterate through render targets", maybe we should loop trough them to see if at least one has explicitly enabled "gamma" and return true at this point. Here you are only checking the first one on the list if I'm not mistaken...

Reply:
![devenv_2018-10-30_16-26-05](https://user-images.githubusercontent.com/1816251/47729505-8f109580-dc60-11e8-9dcb-887e8019bd6b.png)

It is possible to get if the RenderSystem itself uses sRGB gamma correction just by querying `getConfigOptions()` on it. This gives you a map of all the configuration values that were given, but doesn't give you the configuration of the render targets. And they may not match.

Reply:
modified your code to check if render system has gamma correction. If so, it will iterate through all render target, if one of them uses hardwareGammaConversion, it will return true.